internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java

```diff
@@ -183,18 +159,15 @@ public static ControllerClient constructClusterControllerClient(
       String clusterName,
       String discoveryUrls,
       Optional<SSLFactory> sslFactory) {
-    if (!clusterToClientMapContains(clusterName, discoveryUrls)) {
-      addClusterToClientMapEntry(
-          clusterName,
-          discoveryUrls,
-          new ControllerClient(clusterName, discoveryUrls, sslFactory));
-    }
-    return getClusterToClientMapEntry(clusterName, discoveryUrls);
+    return ControllerClientFactory.getControllerClient(clusterName, discoveryUrls, sslFactory);
   }
 
   @Override
   public void close() {
-    clusterToClientMap.clear();
+    if (getShared()) {
+      // Object is still in use at other places. Do not release resources right now.
+      ControllerClientFactory.release(this);
+    }
```

Comment:
I'm wondering if we really need logs here? Depending on the usage of the shared objects, the logs could grow quite a bit

Reply:
I don't expect we will close the `ControllerClient` too often but I could be wrong and it can be helpful for debugging when someone believes the client has been closed but in fact it hasn't.

I'm okay to not add the log. Drop the comment.